src/main/java/org/opentripplanner/middleware/bugsnag/jobs/BugsnagEventJob.java

```diff
@@ -0,0 +1,112 @@
+package org.opentripplanner.middleware.bugsnag.jobs;
+
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Sorts;
+import org.bson.conversions.Bson;
+import org.opentripplanner.middleware.bugsnag.BugsnagDispatcher;
+import org.opentripplanner.middleware.models.BugsnagEvent;
+import org.opentripplanner.middleware.models.BugsnagEventRequest;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.TypedPersistence;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsInt;
+
+/**
+ * Job to check for new event data requests and confirm that Bugsnag has completed the request. At which point obtain
+ * the event data from Bugsnag storage, save to Mongo and remove stale events.
+ */
```

Comment:
I feel like this JavaDoc could be expanded upon a bit. Perhaps some of the detail that was written in the design doc can be included in here.

Reply:
Updated.